Implement APIBlueprintWriter

cilim opened the pull request:
Hello API Hogs company 👋 

I love your library and I created a writer that supports API Blueprint markdown generation.

I've been using this library with my writer for some time now on a couple of production apps and I think it would be awesome if you could review this pull request and hopefully merge it to master so everyone can benefit from it. 😃 

Regards!

mbuhot commented on mix.exs:
```diff
@@ -30,7 +30,8 @@ defmodule Bureaucrat.Mixfile do
      {:phoenix, "~> 1.2", optional: true},
-     {:ex_doc, ">= 0.0.0", only: :dev}
+     {:ex_doc, ">= 0.0.0", only: :dev},
+     {:inflex, "~> 1.10.0"}
```
Where is inflex used?

cilim replied:
Here: https://github.com/api-hogs/bureaucrat/pull/34/files#diff-d36f774419f37ac24b6ba39063685d10R149

mbuhot commented on lib/bureaucrat/helpers.ex:
```diff
@@ -109,4 +109,12 @@ defmodule Bureaucrat.Helpers do
     end
   end
+
+  def plug_doc(conn, module: module, action: action) do
```
A `@doc` string on this one will be helpful.

mbuhot commented on README.md:
```diff
@@ -200,4 +233,5 @@ be written to `web/controllers/api/v1/README.md`.
 * `:titles`: Allows you to specify explicit titles for some of your modules.
 For example `[{YourApp.Api.V1.UserController, "API /v1/users"}]` will
 change the title (Table of Contents entry and heading) for this controller.
+* `:prefix`: Allows you to remove the prefix of the test module names
```
Where is this config used?

cilim replied:
Here: https://github.com/api-hogs/bureaucrat/pull/34/files#diff-d36f774419f37ac24b6ba39063685d10R144

cilim commented:
@mbuhot I accidentally pushed the wrong markdown file in the first commit. Thanks for reviewing the PR.

I just corrected the markdown writer and added the @doc to the `plug_doc()` function.

OpakAlex commented on lib/bureaucrat/api_blueprint_writer.ex:
```diff
@@ -0,0 +1,159 @@
+
+    file |> puts("### #{test_description} [#{method} #{anchor(record_request)}]")
+
+    unless path_params == %{} do
```
can we extract this to function, and maybe use pattern matching we can remove unless. Like,
```elixir
defp write_paramters(_path_params =%{}, file) do
  nil
end
defp write_parameters(path_params, file)
file
     |> puts("\n+ Parameters\n#{formatted_params(path_params)}")

      Enum.each(path_params, fn {param, value} ->
        puts(file, indent_lines(12, "#{param}: #{value}"))
      end)
end
```

OpakAlex commented on lib/bureaucrat/api_blueprint_writer.ex:
```diff
@@ -0,0 +1,159 @@
+    |> puts("\n\n+ Request #{record.assigns.bureaucrat_desc}")
+    |> puts("**#{record.method}**&nbsp;&nbsp;`#{path}`\n")
+
+    unless record.req_headers == [] do
```
same refactor can be done here.

OpakAlex commented on lib/bureaucrat/api_blueprint_writer.ex:
```diff
@@ -0,0 +1,159 @@
+      file
+    end
+
+    unless record.body_params == %{} do
```
same

OpakAlex commented on lib/bureaucrat/api_blueprint_writer.ex:
```diff
@@ -0,0 +1,159 @@
+    file
+    |> puts("\n+ Response #{record.status}\n")
+
+    unless record.resp_headers == [] do
```
same

cilim commented:
@OpakAlex Totally forgot about those Ruby leftovers xD
Thanks for the review :)

PR is updated!

cilim commented:
Awesome! Let me know when you make a new release :)
